### ISSUE TITLE
`Style/CaseLikeIf` Error

### ISSUE THREAD

**Issue by a-lavis** [rubocop/rubocop#12690]
## Expected behavior

When running `rubocop foo.rb` where `foo.rb` is the following file:
```ruby
# frozen_string_literal: true

def calculate(foo)
  if foo == 1
    'one!'
  elsif foo == 2
    'two!'
  else
    foo == 3 ? 'three!' : 'other number'
  end
end

puts calculate(9)
```

I expect RuboCop to not encounter an error.

## Actual behavior

When running `rubocop foo.rb`, I see the following output:
```
Inspecting 1 file
An error occurred while Style/CaseLikeIf cop was inspecting /Users/{username}/foo.rb:4:2.
To see the complete backtrace run rubocop -d.
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/CaseLikeIf cop was inspecting /Users/{username}/foo.rb:4:2.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.3.0) [arm64-darwin23]
```

## Steps to reproduce the problem

Run `rubocop foo.rb` where `foo.rb` is the above file.

## RuboCop version

```
> rubocop -V
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.3.0) [arm64-darwin23]
```
